pkg/resources/sku/sku_test.go

```diff
@@ -0,0 +1,633 @@
+						},
+						sku: pointerToSKU,
+					},
+				},
+				isUpdated: false,
+			},
+			wantErr: false,
+			validate: func(sku *SKU, t *testing.T) {
+				gotReplicas := sku.GetProduct(v1alpha1.Product3Scale).GetReplicas("apicast_production")
+				wantReplicas := int32(1)
+				if gotReplicas != wantReplicas {
+					t.Errorf("Expected apicast_production replicas to be '%v' but got '%v'", wantReplicas, gotReplicas)
+				}
+				// to do more interrogations
+			},
+		},
+		{
+			name: "test successful parsing of config map to sku object for TWENTY million SKU",
+			args: args{
+				SKUId:     TWENTYMILLIONSKU,
+				SKUConfig: getSKUConfig(nil),
+				SKU:       pointerToSKU,
+			},
+			want: &SKU{
+				name: TWENTYMILLIONSKU,
+				productConfigs: map[v1alpha1.ProductName]ProductConfig{
+					v1alpha1.Product3Scale: {
+						productName: v1alpha1.Product3Scale,
+						resourceConfigs: getResourceConfig(func(rcs map[string]ResourceConfig) {
+							rcs["backend_listener"] = ResourceConfig{
+								Replicas: int32(3),
+								Resources: v13.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("0.25"),
+										corev1.ResourceMemory: resource.MustParse("450"),
+									},
+									Limits: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("0.3"),
+										corev1.ResourceMemory: resource.MustParse("500"),
+									},
+								},
+							}
+						}),
+						sku: pointerToSKU,
+					},
+					v1alpha1.ProductMarin3r: {
+						productName: v1alpha1.ProductMarin3r,
+						resourceConfigs: map[string]ResourceConfig{
+							"marin3r": {0, v13.ResourceRequirements{}},
+						},
+						sku: pointerToSKU,
+					},
+					v1alpha1.ProductRHSSOUser: {
+						productName: v1alpha1.ProductRHSSOUser,
+						resourceConfigs: map[string]ResourceConfig{
+							"keycloak": {0, v13.ResourceRequirements{}},
+						},
+						sku: pointerToSKU,
+					},
+				},
+				isUpdated: false,
+			},
+			wantErr: false,
+			validate: func(sku *SKU, t *testing.T) {
+				gotReplicas := sku.GetProduct(v1alpha1.Product3Scale).GetReplicas("backend_listener")
+				wantReplicas := int32(3)
+				if gotReplicas != wantReplicas {
+					t.Errorf("Expected apicast_production replicas to be '%v' but got '%v'", wantReplicas, gotReplicas)
+				}
+				// to do more interrogations
+				// check the rate limit amounts
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := GetSKU(tt.args.SKUId, tt.args.SKUConfig, tt.args.SKU)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetSKU() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if tt.want != nil && !reflect.DeepEqual(tt.want, tt.args.SKU) {
+				t.Errorf("they don't match, \n got = %v, \n want= %v ", tt.args.SKU, tt.want)
+			}
+
+			if tt.validate != nil {
+				tt.validate(tt.args.SKU, t)
+			}
+		})
+	}
+}
+
+func TestProductConfig_Configure(t *testing.T) {
```

Comment:
This is some awesome looking unit test, at a glance!

Reply:
It looks like we're only testing deployment configs though? We should check that this also functions for the 3scale CRD, a deployment and a stateful set.

Reply:
We should also check that a cluster with more resources than necessary is lowered when isUpdated is true.

Reply:
~cluster~ I mean deployment, deploymentconfig, statefulset or 3scale CR

Reply:
ah my mistake, I see we are testing all the types, but using the same name on each one :facepalm: still worth testing the isUpdate lowers values anyway

Reply:
ah sorry should have updated the names. I didn't cover 3scale CRD. but I can add that also later. About lowering the values. I have a test that they don't get lowered even when isUpdated is true as I thought we only wanted to update them if they were lower?